src/core/mount.c

```diff
@@ -1592,11 +1603,51 @@ static int mount_get_timeout(Unit *u, usec_t *timeout) {
         return 1;
 }
 
+static void synthesize_root_mount(Manager *m) {
```

Comment:
Shouldn't this return an int? Is seems like a failure here should cascade up to the caller of mount enumerate. And you'd avoid the ugly `{ log_oom(); return; }` combos.